app/dao/report_requests_dao.py

```diff
@@ -0,0 +1,14 @@
+from uuid import UUID
+
+from app import db
+from app.dao.dao_utils import autocommit
+from app.models import ReportRequest
+
+
+@autocommit
+def dao_create_report_request(report_request: ReportRequest):
+    db.session.add(report_request)
+
+
+def dao_get_report_request_by_id(service_id: UUID, report_id: UUID) -> ReportRequest | None:
```

Comment:
Optional, but this won't return `None` if no result is found it will raise an error so `None` could be removed from the type hint just to avoid confusion. Raising an error is fine though and works well with the frontend code.